src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -619,6 +620,28 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### `astro:route:setup`
+
+<p><Since v="4.14.0" /></p>
+
+**When:** In `astro dev`, whenever a route is requested. In `astro build`, while bundling and transforming a route file.
+
+**Why:** To update a route's option, i.e. `prerender` to configure [on-demand server rendering](/en/guides/server-side-rendering/#opting-in-to-pre-rendering-in-server-mode).
+
+```js
+'astro:route:setup'?: (options: { route: RouteOptions }) => void | Promise<void>;
+```
+
+#### `route` option
+
+**Type:** `RouteOptions`
+
+An object with a readonly `component` property and a `prerender` property.
+
+The `component` property is a readonly string path relative to the project root, e.g `"src/pages/blog/[...slug].astro"`.
+
+The `prerender` property is used to configure [on-demand server rendering](/en/guides/server-side-rendering/#opting-in-to-pre-rendering-in-server-mode) for a route. The value may be a `boolean` or `undefined` depending if there's an explicit `export const prerender` value set in the file. If the final value after running all the hooks is `undefined`, it will fall back to a prerender default based on the [`output` option](/en/reference/configuration-reference/#output).
```

Comment:
``````suggestion

#### `route.prerender`

**Type:** `boolean`<br />
**Default:** `undefined`<br />
<Since v="4.14.0" />
</p>
The `prerender` property is used to configure [on-demand server rendering](/en/guides/server-side-rendering/#opting-in-to-pre-rendering-in-server-mode) for a route. 

If the route file contains an explicit `export const prerender` value, ...
Otherwise, ...

```js
function setPrerender() {
  return {
    name: 'set-prerender',
    hooks: {
      'astro:route:setup': ({ route }) => {
        if (route.component.endsWith('/blog/[slug].astro')) {
          route.prerender = true;
        }
      },
    },
  };
}
```
 
If the final value after running all the hooks is `undefined`, the route will fall back to a prerender default based on the [`output` option](/en/reference/configuration-reference/#output): prerendered for `hybrid` mode, and on-demand rendered for `server` mode.
``````

So, make this correct, but I was thinking maybe something like this format style?